src/VirtualClient/VirtualClient.Dependencies/Packaging/SnapPackageInstallation.cs

```diff
@@ -0,0 +1,214 @@
+
+            // Determine which packages should be installed, and which can be skipped.
+            List<string> toInstall = new List<string>();
+            foreach (string package in packages)
+            {
+                if (!this.AllowUpgrades && await this.IsPackageInstalledAsync(package, cancellationToken))
+                {
+                    this.Logger.LogTraceMessage($"Package '{package}' is already installed, skipping.", EventContext.Persisted());
+                }
+                else
+                {
+                    toInstall.Add(package);
+                }
+            }
+
+            // Nothing to install.
+            if (toInstall.Count == 0)
+            {
+                return;
+            }
+
+            string formattedArguments = $"install {string.Join(' ', toInstall)}";
+
+            await this.InstallRetryPolicy.ExecuteAsync(async () =>
+            {
+                // Runs Snap update first.
+                await this.ExecuteCommandAsync(SnapPackageInstallation.SnapCommand, $"refresh", Environment.CurrentDirectory, telemetryContext, cancellationToken).ConfigureAwait(false);
+
+                // Runs the installation command with retries and throws if the command fails after all
+                // retries are expended.
+                await this.ExecuteCommandAsync(SnapPackageInstallation.SnapCommand, formattedArguments, Environment.CurrentDirectory, telemetryContext, cancellationToken).ConfigureAwait(false);
+            }).ConfigureAwait(false);
+
+            this.Logger.LogTraceMessage($"VirtualClient installed Snap package(s): '[{string.Join(' ', toInstall)}]'.", EventContext.Persisted());
+
+            // Then, confirms that the packages were installed.
+            List<string> failedPackages = toInstall.Where(package => !(this.IsPackageInstalledAsync(package, cancellationToken).GetAwaiter().GetResult())).ToList();
+            if (failedPackages?.Count > 0)
+            {
+                throw new ProcessException(
+                    $"Packages were supposedly successfully installed, but cannot be found! Packages: '{string.Join(", ", failedPackages)}'",
+                    ErrorReason.DependencyInstallationFailed);
+            }
+        }
+
+        /// <inheritdoc />
+        protected override bool IsSupported()
+        {
+            bool shouldExecute = false;
+            if (base.IsSupported())
+            {
+                shouldExecute = this.Platform == PlatformID.Unix;
+            }
+
+            return shouldExecute;
+        }
+
+        private async Task SetupSnapdSockets(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            var linuxDistributionInfo = await this.systemManagement.GetLinuxDistributionAsync(cancellationToken)
+                .ConfigureAwait(false);
+
+            switch (linuxDistributionInfo.LinuxDistribution)
+            {
+                case LinuxDistribution.CentOS7:
```

Comment:
For ubuntu do we need anything? If not, could you add a comment here to make it a little more obvious?

Reply:
sorry did you see this one? somehow I'm still not seeing any comment